osu.Game/Skinning/LegacySkin.cs

```diff
@@ -262,17 +262,24 @@ public override Texture GetTexture(string componentName)
 
         public override SampleChannel GetSample(ISampleInfo sampleInfo)
         {
-            foreach (var lookup in sampleInfo.LookupNames)
+            var hsi = sampleInfo as HitSampleInfo;
+
+            if (hsi == null || hsi.IsCustom)
```

Comment:
I don't think this is exactly equivalent to the code before - if an `ISampleInfo` of another type is passed in here it won't have its `LookupNames` checked...?